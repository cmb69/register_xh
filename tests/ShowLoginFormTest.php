<?php

/**
 * Copyright (c) 2021-2023 Christoph M. Becker
 *
 * This file is part of Register_XH.
 */

namespace Register;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Register\Infra\Request;
use Register\Infra\Url;
use Register\Infra\UserRepository;
use Register\Infra\View;
use Register\Value\User;

class ShowLoginFormTest extends TestCase
{
    public function testLoginForm(): void
    {
        $sut = new ShowLoginForm($this->conf(), $this->text(), $this->userRepo(), $this->view());
        $response = $sut($this->request());
        Approvals::verifyHtml($response->output());
    }

    public function testLoggedInOnlyForm(): void
    {
        $sut = new ShowLoginForm([], $this->text(), $this->userRepo(), $this->view());

        $response = $sut($this->request(), true);

        $this->assertEquals("", $response->output());

    }

    public function testLoggedInForm(): void
    {
        $sut = new ShowLoginForm([], $this->text(), $this->userRepo(), $this->view());

        $request = $this->request();
        $request->method("username")->willReturn("jane");
        $response = $sut($request);

        Approvals::verifyHtml($response->output());
    }

    private function request(): Request
    {
        $request = $this->createMock(Request::class);
        $request->method("url")->willReturn(new Url("/", "Foo"));
        return $request;
    }

    private function userRepo(): UserRepository
    {
        $userRepo = $this->createMock(UserRepository::class);
        $userRepo->method("findByUsername")->willReturn(
            new User("jane", "", [], "Jane Doe", "jane@example.com", "activated", "secret")
        );
        return $userRepo;
    }

    private function view(): View
    {
        return new View("./views/", $this->text());
    }

    private function conf(): array
    {
        return XH_includeVar("./config/config.php", "plugin_cf")["register"];
    }

    private function text(): array
    {
        return XH_includeVar("./languages/en.php", "plugin_tx")["register"];
    }
}
