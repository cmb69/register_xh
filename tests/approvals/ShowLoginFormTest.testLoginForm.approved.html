<!-- register login form -->
<div class="register_login">
  <form method="post">
    <input type="hidden" name="function" value="registerlogin">
    <p class="register_field">
      <label>
        <span>Username</span>
        <input name="username">
    </p>
    <p class="register_field">
      <label>
        <span>Password</span>
        <input type="password" name="password">
      </label>
    </p>
    <p class="register_buttons">
      <button name="login">Login</button>
      <label>
        <input type="checkbox" name="remember">
        <span>Remember User</span>
      </label>
    </p>
    <p class="register_links">
      <a href="/?Password-forgotten">Password forgotten</a>
      <a href="/?Register">Register</a>
    </p>
  </form>
</div>
